### ISSUE TITLE
Enable gVisor sandbox as a sandbox alternative (First step)

### ISSUE THREAD

**Issue by nathanieltornow** [ease-lab/vhive#303]
Hi!
We (a group of students at TU Munich) would like to add support for [gVisor](https://gvisor.dev/)-Sandboxes to vHive, which can be used on demand as an alternative for firecracker-VMs.

The first step for this would be following:

### Refactoring and generalising cri
Overall layout:
```txt
cri/
    service.go 
    proxy.go
    container_create.go
    container_remove.go
    funcInstance.go
    coordinator/gvisor/coordinator.go
    coordinator/firecracker/coordinator.go
ctriface/firecracker/
    # same as ctriface before
ctriface/gvisor
    # api implemetation for gvisor sandboxes
```

Description:
- Creating a generic interface for a coordinator:
  ```go
  type Coordinator interface {
      // the functionality of coordinator.startVM
      StartSandbox(context.Context, string) (*FuncInstance, error)
      // // the functionality of coordinator.stopVM
      StopSandbox(context.Context, string) error
  }
  ```
  StartSandbox and StopSandbox are the same as coordinator.startVM and coordinator.stopVM for the firecracker-coordinator. 

- Changing cri.Service to call the functions of the Coordinator interface.
   cri.NewService will not take and Orchestrator-type as it's argument (the only usage for that orchestrator is to hand it over to the coordinator anyway), but a type that implements the Coordinator interface (either a gvisor or a firecracker coordinator).
   
    The Firecracker Coordinator:
   ```go
   type FirecrackerCoordinator struct {
       Coordinator
       //.... Same functionality as before
   }
   
   func (fcCoor *FirecrackerCoordinator)StartSandbox(ctx context.Context, image string) (*FuncInstance, error) {
       return startVM(ctx, image) // the same function as right now
   }
   
   func (fcCoor *FirecrackerCoordinator)StopSandbox(ctx context.Context, id string) error {
       return stopVM(ctx, id) // the same function as right now
   }
   ```
   
   The new Service:
   ```go
   type Service struct {
       sync.Mutex
       criapi.ImageServiceServer
       criapi.RuntimeServiceServer
       stockRuntimeClient criapi.RuntimeServiceClient
       stockImageClient   criapi.ImageServiceClient
       coordinator        *Coordinator // generic coordinator
   }
   
   func NewService(coor *Coordinator) (*Service, error) {
       // same as before (without orchestrator)
   }
   ```
   
  Creating a new firecracker-cri-service will look as follows:
   ```go
   orch := ctriface.NewOrchestrator(...)
   coordinator := fcCoor.NewFirecrackerCoordinator(orch)
   service := cri.NewService(coordinator) 
   ```

**Comment by nathanieltornow** [comment]
We will hand in a PR for these changes asap, which won't change/ or add any functionality yet (only refactoring)

**Comment by ustiugov** [comment]
fixed by #389
